examples/typescript/src/App.tsx

```diff
@@ -21,46 +32,40 @@ const filters: ChannelFilters = { type: 'messaging', members: { $in: [userId] }
 const options: ChannelOptions = { state: true, presence: true, limit: 10 };
 const sort: ChannelSort = { last_message_at: -1, updated_at: -1 };
 
-type LocalAttachmentType = Record<string, unknown>;
-type LocalChannelType = Record<string, unknown>;
-type LocalCommandType = string;
-type LocalEventType = Record<string, unknown>;
-type LocalMessageType = Record<string, unknown>;
-type LocalReactionType = Record<string, unknown>;
-type LocalUserType = Record<string, unknown>;
-
-type StreamChatGenerics = {
-  attachmentType: LocalAttachmentType;
-  channelType: LocalChannelType;
-  commandType: LocalCommandType;
-  eventType: LocalEventType;
-  messageType: LocalMessageType;
-  reactionType: LocalReactionType;
-  userType: LocalUserType;
-  pollType: UR;
-  pollOptionType: UR;
-};
-
-const chatClient = StreamChat.getInstance<StreamChatGenerics>(apiKey);
+const App = () => {
+  const chatClient = useCreateChatClient({
+    apiKey,
+    tokenOrProvider: userToken,
+    userData: { id: userId, customUserField: 'field' },
+  });
```

Comment:
Note how `userData` supports `customUserField` and doesn't cause type error. That's because `useCreateChatClient` uses `UserResponseEx` type for it.